src/core/js/views/componentView.js

```diff
@@ -6,15 +6,17 @@ define([
     var ComponentView = AdaptView.extend({
 
         className: function() {
-            return "component " + 
-            this.model.get('_component') + 
-            "-component " + this.model.get('_id') + 
-            " " + this.model.get('_classes') + 
-            " " + this.setVisibility() +
-            " " + this.setHidden() +
-            " component-" + this.model.get('_layout') + 
-            " nth-child-" + this.model.get("_nthChild") +
-            " " + (this.model.get('_isComplete') ? 'completed' : '');
+            return [
+                'component',
+                this.model.get('_component') +'-component',
+                this.model.get('_id'),
+                this.model.get('_classes'),
+                this.setVisibility(),
+                this.setHidden(),
+                'component-' + this.model.get('_layout'),
+                'nth-child-' + this.model.get('_nthChild'),
```

Comment:
I think we can start to deprecate the custom nth-child classes in favor to native css selectors.